tests/test_resource_base.py

```diff
@@ -32,3 +32,47 @@ def test_validate_resource_id_custom_message(client):
         resource.validate_resource_id("invalid", message="No no no, only tests allowed here!")
 
     assert str(excinfo.value) == "No no no, only tests allowed here!"
+
+
+def test_response_handling_error_sets_message_and_key_correctly():
+    try:
+        raise ResponseHandlingError("This is a ResponseHandlingError", "test_idempotency_key")
+    except ResponseHandlingError as exc:
+        assert exc.idempotency_key == "test_idempotency_key"
+        assert exc.args[0] == "This is a ResponseHandlingError"
```

Comment:
If some code raises an exception, you never consult `args[0]` while handling the exception, so why test it? Could you test this the same way as all the other tests: `assert str(excinfo.value) == "..."`?